### ISSUE TITLE
Can't get it work

### ISSUE THREAD

**Issue by lgmorand** [Azure/kaito#344]
**Describe the bug**
workspace is never ready

```
NAME                            INSTANCE            RESOURCEREADY   INFERENCEREADY   WORKSPACEREADY   AGE
workspace-falcon-7b-instruct    Standard_NC6s_v3    False                            False            33m
workspace-mistral-7b-instruct   Standard_NC12s_v3   False                            False            85m
```

**Steps To Reproduce**
First I wanted to follow the [documentation](https://learn.microsoft.com/en-us/azure/aks/ai-toolchain-operator) (but links to yaml file are no broken), but when fixed I still can't get any workspace or resource ready.

For the NC12s, I thought it could be a quota. but for NC6, I even added a nodepool to check and it's fine.

**How can I get some logs on what's is going wrong ?**

Last point: I see that Kaito is now 2.2 here on GH but I'm using the AKS addon. could it use an oldest broken version ?

**Comment by Fei-Guo** [comment]
You can check the machine CR and see the status. `kubectl get machines -o yaml`. Usually if the workspace stuck at waiting for resource ready, this means the GPU VM creation hits error (mostly quota issues). You can check the machine CR for the NC6 node for more detailed error. 

Note that AKS addon is running 0.1.0 version, which does not include mistral model. https://github.com/Azure/kaito/tree/main/presets/README.md lists the supported models per kaito release.

**Comment by lgmorand** [comment]
Thanks Fei

Here is the yaml of machine. it does not seem to like my template (which I gave below)

```
apiVersion: v1
items:
- apiVersion: karpenter.sh/v1alpha5
  kind: Machine
  metadata:
    creationTimestamp: "2024-04-08T20:58:32Z"
    finalizers:
    - karpenter.sh/termination
    generation: 1
    labels:
      apps: falcon-7b-instruct
      kaito.sh/workspace: workspace-falcon-7b-instruct
      kaito.sh/workspacenamespace: default
      karpenter.sh/provisioner-name: default
    name: wsffcd1580e
    resourceVersion: "6288"
    uid: e21d797e-786f-411d-a75c-86c5009158a0
  spec:
    machineTemplateRef:
      name: wsffcd1580e
    requirements:
    - key: node.kubernetes.io/instance-type
      operator: In
      values:
      - Standard_NC6s_v3
    - key: karpenter.sh/provisioner-name
      operator: In
      values:
      - default
    - key: kaito.sh/machine-type
      operator: In
      values:
      - gpu
    - key: kubernetes.io/arch
      operator: In
      values:
      - amd64
    - key: kubernetes.io/os
      operator: In
      values:
      - linux
    resources:
      requests:
        storage: 50Gi
    taints:
    - effect: NoSchedule
      key: sku
      value: gpu
  status:
    conditions:
    - lastTransitionTime: "2024-04-08T20:58:38Z"
      message: Machine is not launched
      reason: MachineNotLaunched
      status: "False"
      type: MachineInitialized
    - lastTransitionTime: "2024-04-08T21:08:31Z"
      message: |-
        creating instance, agentPool.BeginCreateOrUpdate for "wsffcd1580e" failed: PUT https://management.azure.com/subscriptions/735a4022-4b84-41dc-8200-dc028954e80d/resourceGroups/rg-demo-aks-kaito/providers/Microsoft.ContainerService/managedClusters/aks-kaito/agentPools/wsffcd1580e
        --------------------------------------------------------------------------------
        RESPONSE 400: 400 Bad Request
        ERROR CODE: PreconditionFailed
        --------------------------------------------------------------------------------
        {
          "code": "PreconditionFailed",
          "details": null,
          "message": "Provisioning of resource(s) for Agent Pool wsffcd1580e failed. Error: {\n  \"code\": \"InvalidTemplateDeployment\",\n  \"message\": \"The template deployment 'f7f018c4-bb35-4327-94c0-ba134c3e2d30' is not valid according to the valid...
      reason: LaunchFailed
      status: "False"
      type: MachineLaunched
    - lastTransitionTime: "2024-04-08T20:58:38Z"
      message: Machine is not launched
      reason: MachineNotLaunched
      status: "False"
      type: MachineRegistered
    - lastTransitionTime: "2024-04-08T21:08:31Z"
      message: Machine is not launched
      reason: MachineNotLaunched
      status: "False"
      type: Ready
kind: List
metadata:
  resourceVersion: ""
```


```
apiVersion: kaito.sh/v1alpha1
kind: Workspace
metadata:
  name: workspace-falcon-7b-instruct
resource:
  instanceType: "Standard_NC6s_v3"
  labelSelector:
    matchLabels:
      apps: falcon-7b-instruct
inference:
  preset:
    name: "falcon-7b-instruct"
```

**Comment by Fei-Guo** [comment]
@lgmorand,  can you please check the gpu-provisioner controller log and past the full err msg here? The msg in the status is truncated. 

-----------------------------------------
  creating instance, agentPool.BeginCreateOrUpdate for "wsffcd1580e" failed: PUT https://management.azure.com/subscriptions/735a4022-4b84-41dc-8200-dc028954e80d/resourceGroups/rg-demo-aks-kaito/providers/Microsoft.ContainerService/managedClusters/aks-kaito/agentPools/wsffcd1580e
        --------------------------------------------------------------------------------
        RESPONSE 400: 400 Bad Request
        ERROR CODE: PreconditionFailed
        --------------------------------------------------------------------------------
        {
          "code": "PreconditionFailed",
          "details": null,
          "message": "Provisioning of resource(s) for Agent Pool wsffcd1580e failed. Error: {\n  \"code\": \"InvalidTemplateDeployment\",\n  \"message\": \"The template deployment 'f7f018c4-bb35-4327-94c0-ba134c3e2d30' is not valid according to the valid...
------------------------------------

I want to understand which Precondition check fails. 

Thanks

**Comment by lgmorand** [comment]
I'm always using the same exact script.

1- I could not get the logs because the cluster was stopped when I restarted it, it was stuck with "Operation in progress (starting"

2- I deleted the cluster and recreated it and now I had a quota issue
" \\\"Operation could not be completed as it results in exceeding approved standardNCSv3Family Cores quota. Additional details - Deployment Model: Resource Manager, Location: westeurope, Current Limit: 6, Current Usage: 6, Additional Required: 6, (Minimum) New Limit Required: 12."
I thought it was counting GPU but now quota is per vCPU

3- I created a new cluster  and then it workds
Then few minutes later, without touching anything. it works. I'll monitor the behavior because I don't change anything on each run.
